core/txpool/blobpool/blobpool.go

```diff
@@ -390,14 +399,52 @@ func (p *BlobPool) Init(gasTip uint64, head *types.Header, reserver txpool.Reser
 	}
 	p.head, p.state = head, state
 
+	// Create new slotter for pre-Osaka blob configuration.
+	slotter := newSlotter(eip4844.LatestMaxBlobsPerBlock(p.chain.Config()))
+
+	// Check if we need to migrate our blob db to the new slotter.
+	if p.chain.Config().IsOsaka(head.Number, head.Time) {
+		// Open the store using the version slotter to see if any version has been
+		// written.
+		var version int
+		index := func(_ uint64, _ uint32, blob []byte) {
+			version = max(version, parseSlotterVersion(blob))
+		}
+		store, err := billy.Open(billy.Options{Path: queuedir}, newVersionSlotter(), index)
+		if err != nil {
+			return err
+		}
+		store.Close()
+
+		// If the version found is less than the currently configured store version,
+		// perform a migration then write the updated version of the store.
+		if version < storeVersion {
+			newSlotter := newSlotterEIP7594(eip4844.LatestMaxBlobsPerBlock(p.chain.Config()))
+			if err := billy.Migrate(billy.Options{Path: queuedir, Repair: true}, slotter, newSlotter); err != nil {
```

Comment:
It's a bit awkward. 

Before the migration, the blob transactions may still still carry the v0 sidecar. We should make the conversion first, and then migrate them to the new shelves respectively.

Reply:
This is done in [0543193](https://github.com/ethereum/go-ethereum/pull/31966/commits/05431933835edf44c7141c25d3186fb0f7bf7ba9), PTAL!